src/main/java/seedu/address/model/person/ModuleCode.java

```
@@ -0,0 +1,55 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a module code that a Person is taking in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidModuleCode(String)}
+ */
+public class ModuleCode {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Module codes should only contain alphanumeric characters, and it should not be blank";
+
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}]*";
```

Currently, the following still works but I think it should throw an error.
`add n/Ben e/ben123@gmail.com m/`

Can we change it to: 
```
public static final String VALIDATION_REGEX = "@[\\p{Alnum}]*[^\\s]";
```

This should reject any whitespace module code inputs.

---

I think using `"[\\p{Alnum}]+"` works too, the  `+` ensures that there is at least 1 alphanumeric character present.